app/models/node.rb

```diff
@@ -102,11 +185,183 @@ class << self
     def root
       Node.find(0)
     end
+
+    def table
+      arel_table
+    end
+
+    # returns (base_id, ...node)
+    def relatives_via_arel_mgr(direction_towards_root, base_node_id = nil)
```

Comment:
probs worth replacing with ARHQ logic if we don't have a reason to have all of this. is a good reference tho